aws_signing_helper/signer.go

```diff
@@ -747,5 +741,18 @@ func ReadCertificateData(certificateId string) (CertificateData, error) {
 	}
 
 	//return struct
-	return CertificateData{keyType, encodedDer, serialNumber, supportedAlgorithms}, nil
+	return CertificateData{keyType, encodedDer, serialNumber, supportedAlgorithms}, cert, nil
+}
+
+// GetCertChain Read a certificate bundle and return a chain of all the certificates is contains
+func GetCertChain(certificateBundleId string) ([]*x509.Certificate, error) {
```

Comment:
Also minor, but maybe it would be good to be consistent with the way in which we handle the parsing a single certificate vs. multiple certificates from a file? For a single certificate, I noticed that you added another return value to the existing `ReadCertificateData` function, but for multiple certificates, you created a new function. Maybe there's a good reason to do it this way though.

Reply:
`ReadCertificateData` already had the information that I needed and was discarding it so it seemed easiest to just return the info so that everywhere that called it didn't have to didn't have to have an extra 8 lines to redecode and reparse the cret data and the caller could chose to use or ignore the new return value.

I added `GetCertChain` to encapsulate the process of appending all the certs from the bundle so that looping didn't need to be repeated when it is called.

If you want I can revert `ReadCertificateData` and create a `GetCertificate` that calls `ReadCertificateData` and then does the decoding and parsing.  Up to you, just let me know.